### DISCUSSION TITLE
preg_replace broken in PHP 8.2

### DISCUSSION THREAD

**Discussion by busybit** [oerdnj/deb.sury.org#1968] [General]
I use the following statement to convert a wiki text into html:
`$content = preg_replace("/'''(.*)'''/Um", '<strong>$1</strong>', $content);`

With content like `'''Text'''` it should give `<strong>Text</strong>`.

This worked with PHP 7.4 from original Debian 11 package, but not with PHP 8.2 from packages.sury.org.

```
php8.2:
  Installed: 8.2.7-1+0~20230609.23+debian11~1.gbp21561a
  Candidate: 8.2.7-1+0~20230609.23+debian11~1.gbp21561a
  Version table:
 *** 8.2.7-1+0~20230609.23+debian11~1.gbp21561a 500
        500 https://packages.sury.org/php bullseye/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by oerdnj** [comment]
This is not a packaging issue. PHP 8.2 uses pcre2, so perhaps you might want to check this with pcre2 upstream.

**Comment by busybit** [comment]
There is nothing wrong with this regex. I wrote a little test, and PHP CLI matched the same pattern as expected. But through PHP FPM the same pattern is not matched.

**Comment by sleemanj** [comment]
Have you tried disabling pcre.jit ini setting?Message ID: ***@***.***>

**Comment by busybit** [comment]
> Have you tried disabling pcre.jit ini setting?Message ID: ***@***.***>

I tried it, but it makes no difference.

**Comment by oerdnj** [comment]
I understand, but yet again, it's hardly a packaging problem. The pcre version is same for both CLI and FPM.

**Comment by busybit** [comment]
But then what could be the problem? The regular expression is fine. Both the PHP 8.2 CLI version matches it correctly, so does the pcr2test utility. Only the FPM version does not match it. In the wiki there are many other replacements that all work, only 3 replacements that replace such single quotes do not work.
